redbot/core/bank.py

```diff
@@ -563,6 +591,10 @@ async def bank_prune(bot: Red, guild: discord.Guild = None, user_id: int = None)
             if user_id in bank_data:
                 del bank_data[user_id]
 
+    payload = BankPruneInformation(scope, tmp)
```

Comment:
`tmp` is only defined when `user_id is None`. This errors if a particular user is pruned.
(I do think the conditionally defined variables in this function are smelly, making this sort of error very easy to make)